### DISCUSSION TITLE
React copy on every page.js build

### DISCUSSION THREAD

**Discussion by p150** [vercel/next.js#14302] [Help]
Hi guys, I am working on a fairly big project and while I was going through the build chunks to see if I can prune any of the unnecessary stuff, I observed a pretty big part of chunks containing the React and React DOM code on both _app.js and particular-page.js. Is this a known issue or is there a way to solve this? It would be great if I can save around 40 KB for my users. Unfortunately I am working on a proprietary codebase and hence can't share the code. But I can try to stitch up a new app to demonstrate the behaviour. Any help is much appreciated. Thank you.

**Comment by cristiand391** [comment]
It has been reported [here](https://github.com/vercel/next.js/issues/12964)

AFAICT, if you are using functional components and not using the `React` variable, you can safely remove the imports. I'm not sure about class components tho.

**Reply by p150** [reply]
Thanks, it's a great starter. Will see if I can make mine work.
